src/llm_vm/onsite_llm.py

```diff
@@ -527,3 +485,56 @@ def finetune(self, dataset, optimizer, c_id, small_model_filename=None):
         # optimizer.storage.set_training_in_progress(c_id, False)
         # if old_model is not None:
         #     openai.Model.delete(old_model)
+
+
+class BaseCtransformers(BaseOnsiteLLM):
+    """
+    Base Class for running Ctransformers/GGML models
+
+    Attributes:
+        model_uri (str): Hugging Face Endpoint for LLM
+        model_kwargs (dict): Keyword arguments for loading the LLM
+    
+    Methods:
+        generate: Generates a response from given prompt
+          
+    """
+    def __init__(self, model_uri=None, model_kwargs={}):
+        if model_uri != None :
+            self.model_uri= model_uri
+        else: 
+            raise ValueError("model_uri not found")
+
+        self.model_file = model_kwargs.pop('model_file', None)
+        self.model = self.model_loader(*model_kwargs)
+
+    def load_finetune(self, model_filename):
+            raise Exception("Finetuning not supported for Ctransformers/GGML.")
+        
+    def generate(self, prompt, *generate_kwargs):  
+        input_ids = self.model.tokenize(prompt)
+        response = self.model.generate(input_ids, *generate_kwargs)
+        return self.model.detokenize(response)
+    
+    def finetune(self, data, optimizer, c_id, model_filename=None   ):
+        raise Exception("Finetuning not supported for Ctransformers/GGML.")
+
+
+@RegisterModelClass("quantized-llama")
+class Quantized_Llama(BaseCtransformers):
+    """
+    Class for running quantized Llama instance
+
+    Methods:
+        model_loader: Loads the LLM into memory
+    """
+
+    model_uri="TheBloke/LLaMa-7B-GGML"
```

Comment:
Which model file is it picking ?

Reply:
the model file is set when initializing the base class. it's passed in with the `model_kwargs`